R/gh.R

```diff
@@ -177,7 +177,20 @@ gh <- function(endpoint, ..., per_page = NULL, .token = NULL, .destfile = NULL,
   while (!is.null(.limit) && length(res) < .limit && gh_has_next(res)) {
     if (.progress) update_progress_bar(prbr, res)
     res2 <- gh_next(res)
-    res3 <- c(res, res2)
+
+    if (!is.null(names(res2)) && identical(names(res), names(res2))) {
+      res3 <- mapply(           # Handle named array case
+        function(x, y) {        # e.g. GET /search/repositories
+          z <- c(x, y)
+          if (is.atomic(z)) unique(z)
```

Comment:
Not sure this is necessary, but it seemed the cleanest option based on the relevant endpoints

Reply:
When do we actually need this?

Reply:
The results from the search api look the following:
```
{
  "total_count": 40,
  "incomplete_results": false,
  "items": [
    {
      ...
    }
  ]
}
```

when there are multiple pages the `items` will change but `total_count` and `incomplete_results` are the same for all of the pages, hence collapsing them with unique. It seems like for this case of response the non-array values are meant to give overall context, but this is purely supposition on my part.